### DISCUSSION TITLE
Enhance create item drawer, and use more screen space

### DISCUSSION THREAD

**Discussion by jooola** [keystonejs/keystone#5455] [Feature requests]
I propose to change some basic admin UI layout problems.

![image](https://user-images.githubusercontent.com/19195485/114550969-17cdf100-9c63-11eb-9f6e-a3c9afd9f45c.png)

Would it be possible to use the whole screen for the drawer instead of squeezing everything in a small column on the right ?

I feel it is wasting space, and some people might want to write a long text for a blog post for example. To do that the only way is to save the item and open the editing page again.

Having a small drawer is probably useful for things like tags, or anything that does not require a long time to create/edit.
But when writing a long blog post you want to focus on that, so having the drawer take the whole page seem like a good idea.

Basically the 'time spend / screen space used / information squeezing' ratio is not good.

What about having a modal looking like this ?
![image](https://user-images.githubusercontent.com/19195485/114552297-bd359480-9c64-11eb-8ff8-2392d6d06b99.png)
Small screen :
![image](https://user-images.githubusercontent.com/19195485/114552576-08e83e00-9c65-11eb-8b8c-da3b25768ebb.png)

Maybe all this could be put in some field options, like small modal for some tags/categories and full screen modal for blog posts ?

I think this has already been discussed some time ago in keystone 5.

EDIT: Fix passive aggressive wording, sorry...

**Comment by bladey** [comment]
Thank you for the great suggestion @jooola and providing so much detail.

**Reply by antonlyap** [reply]
Is there any hope of having this implemented in the near future?
